### PR TITLE
Step-47: unify code

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: examples/step-47/step-47.cc [hunk 1]
```diff
@@ -837,23 +837,27 @@ namespace Step47
...
+                local_error +=
+                  ((exact_hessians[q_point] - hessians[q_point]).norm_square() *
+                   fe_values.JxW(q_point));
+              }
+            error_per_cell[cell->active_cell_index()] = std::sqrt(local_error);
+          }
 
-      const double error_norm = error_per_cell.l2_norm();
+      const double error_norm =
+        VectorTools::compute_global_error(triangulation,
```
Comment (peterrum) [inline examples/step-47/step-47.cc:858]: Just like in the two instances above.

Comment (jppelteret) [reply examples/step-47/step-47.cc:858]: Seems reasonable to me.

**Review by jppelteret — APPROVED** [review]

File: examples/step-47/step-47.cc [hunk 2]
```diff
@@ -837,23 +837,27 @@ namespace Step47
...
-          double local_error = 0;
-          for (unsigned int q_point = 0; q_point < n_q_points; ++q_point)
-            {
-              local_error +=
-                ((exact_hessians[q_point] - hessians[q_point]).norm_square() *
-                 fe_values.JxW(q_point));
-            }
-          error_per_cell[cell->active_cell_index()] = std::sqrt(local_error);
-        }
+        if (cell->is_locally_owned())
```
Comment (tjhei) [inline examples/step-47/step-47.cc:840]: The code is sequential, why do you introduce this check?

Comment (peterrum) [reply examples/step-47/step-47.cc:840]: I can remove this check. But, I think it does not harm here: on the one hand these lines of code mimic in a certain sense `VT::do_integrate_difference()` (the result is not accumulated direct into a temporal variable but into a vector, which is accessed by `cell->active_cell_index()`) and on the other hand this one less change one has to perform if that person wants to parallelize the code (which already happened at least 2 times of which I am aware of).

Comment (tjhei) [reply examples/step-47/step-47.cc:840]: I don't feel strongly about it, but I think we avoid changes like this.
Anybody else have an opinion here?

Comment (peterrum) [reply examples/step-47/step-47.cc]: I have removed it.

Comment (jppelteret) [reply examples/step-47/step-47.cc]: I was going to suggest that we follow @tjhei's opinion here. I'm glad that it's now a non-issue.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**